controllers/wildflyserver_controller_test.go

```diff
@@ -494,3 +495,310 @@ func TestWildFlyServerWithResources(t *testing.T) {
+	assert.Equal("/health/ready", stsReadinessProbe.HTTPGet.Path)
+	assert.Equal("admin", stsReadinessProbe.HTTPGet.Port.StrVal)
+
+	stsStartupProbe := statefulSet.Spec.Template.Spec.Containers[0].StartupProbe
+	assert.Equal(int32(30), stsStartupProbe.InitialDelaySeconds)
+	assert.Equal(int32(345), stsStartupProbe.TimeoutSeconds)
+	assert.Equal(int32(35), stsStartupProbe.PeriodSeconds)
+	assert.Equal(int32(31), stsStartupProbe.SuccessThreshold)
+	assert.Equal(int32(39), stsStartupProbe.FailureThreshold)
+	assert.Equal("/health/live", stsStartupProbe.HTTPGet.Path)
+	assert.Equal("admin", stsStartupProbe.HTTPGet.Port.StrVal)
+}
+
+func TestWildFlyServerWithProbesScript(t *testing.T) {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	assert := testifyAssert.New(t)
+
+	// First verify the default values are created when there is not
+	// any Probe configuration
+	wildflyServer := &wildflyv1alpha1.WildFlyServer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: wildflyv1alpha1.WildFlyServerSpec{
+			ApplicationImage: applicationImage,
+			Replicas:         replicas,
+		},
+	}
+	// Objects to track in the fake client.
+	objs := []runtime.Object{
+		wildflyServer,
+	}
+
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+	s.AddKnownTypes(wildflyv1alpha1.GroupVersion, wildflyServer)
+
+	// Create a fake client to mock API calls.
+	cl := fake.NewClientBuilder().WithRuntimeObjects(objs...).Build()
+
+	// Create a WildFlyServerReconciler object with the scheme and fake client.
+	r := &WildFlyServerReconciler{
+		Client: cl,
+		Scheme: s,
+		Log:    ctrl.Log.WithName("test").WithName("WildFlyServer"),
+	}
+
+	// Mock request to simulate Reconcile() being called on an event for a
+	// watched resource .
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	livenessScriptName := "test-liveness-script.sh"
+	readinessScriptName := "test-readiness-script.sh"
+	os.Setenv("SERVER_LIVENESS_SCRIPT", livenessScriptName)
+	os.Setenv("SERVER_READINESS_SCRIPT", readinessScriptName)
+	// Creating StatefulSet and loop waiting for reconcile to end
+	_, err := reconcileUntilDone(t, r, req, 4)
+
+	// Check if stateful set has been created and has the correct size.
+	statefulSet := &appsv1.StatefulSet{}
+	err = cl.Get(context.TODO(), req.NamespacedName, statefulSet)
+	require.NoError(t, err)
+	assert.Equal(replicas, *statefulSet.Spec.Replicas)
+	assert.Equal(applicationImage, statefulSet.Spec.Template.Spec.Containers[0].Image)
+
+	// check the Probes values
+	stsLivenessProbe := statefulSet.Spec.Template.Spec.Containers[0].LivenessProbe
+	assert.Equal(int32(60), stsLivenessProbe.InitialDelaySeconds)
+	assert.Equal(int32(0), stsLivenessProbe.TimeoutSeconds)
+	assert.Equal(int32(0), stsLivenessProbe.PeriodSeconds)
+	assert.Equal(int32(0), stsLivenessProbe.SuccessThreshold)
+	assert.Equal(int32(0), stsLivenessProbe.FailureThreshold)
```

Comment:
> // Defaults to 3. Minimum value is 1.

how this can then be equal 0?

Reply:
So to wrap it up as only `InitialDelaySeconds` is allowed to have 0 a we redefine default for that one it is safe for other properties to be initialized with 0 with meaning as uninitialized - ok then.

Reply:
Yes, right.